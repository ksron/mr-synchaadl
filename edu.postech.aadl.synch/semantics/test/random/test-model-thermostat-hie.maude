
set show timing off .

load ../../interpreter-random.maude
load ../syntax/module-thermostat-hie.maude


---------------------------------------------
mod TEST-THERMOSTAT-SYSTEM-HIE is
  including THERMOSTAT-SYSTEM-SYNTAX .
  including MODEL-TRANSITION-SYSTEM .
  
  op initState : -> Object .
  eq initState = initialize(model) .

  op cinitState : -> Object .
  eq cinitState = initialize(collapse(model)) .
endm


rew [100] {{initState,0} | 0 | infty} .

rew [100] {{initState,400} | 0 | infty} .

rew [100] {{initState,800} | 0 | infty} .

rew [100] {{initState,1200} | 0 | infty} .


rew [100] {{cinitState,0} | 0 | infty} .

rew [100] {{cinitState,400} | 0 | infty} .

rew [100] {{cinitState,800} | 0 | infty} .

rew [100] {{cinitState,1200} | 0 | infty} .
