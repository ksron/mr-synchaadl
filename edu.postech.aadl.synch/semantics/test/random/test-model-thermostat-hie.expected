==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{initState,0} | 0 | infty} .
rewrites: 75359
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< csys : System | properties :
    none,features : (< on : DataOutPort | properties : none,content : * # [
    false] > < off : DataOutPort | properties : none,content : * # [false] > <
    temp : DataInPort | properties : none,content : [0.0] # [false],cache : [
    0.0] # [false] >),subcomponents : < cproc : Process | properties : none,
    features : (< on : DataOutPort | properties : none,content : * # [false] >
    < off : DataOutPort | properties : none,content : * # [false] > < temp :
    DataInPort | properties : none,content : [0.0] # [false],cache : [0.0] # [
    false] >),subcomponents : < cthread : Thread | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::SamplingTime => {{
    0.0 .. 2.0}}) ; HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features :
    (< on : DataOutPort | properties : none,content : * # [false] > < off :
    DataOutPort | properties : none,content : * # [false] > < temp : DataInPort
    | properties : none,content : [0.0] # [false],cache : [
    9.4249841706512953e+1] # [true] >),subcomponents : (< lower : Data |
    properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    9.3596867381845982e+1] # [true] >),connections : empty,varGen : <
    "top.cthread" >,currState : init,completeStates : init,variables : x :
    Real,transitions : init -[on dispatch]-> init {(v{x} := ((c[curr] + f[
    temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !)
    end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >,connections : ((
    temp --> (cthread .. temp)) ; ((cthread .. on) --> on) ; (cthread .. off)
    --> off) >,connections : ((temp --> (cproc .. temp)) ; ((cproc .. on) -->
    on) ; (cproc .. off) --> off) > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : csys > < off :
    EnvInPort | properties : none,content : * # [false],envCache : null(Unit),
    target : csys > < out : EnvOutPort | properties : none,content : [0.0] # [
    false],envCache : null(Real),target : csys >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [9.5200193572053749e+1] #
    [true] >,connections : x ==> out,varGen : < "top.environment" >,currMode :
    init,jumps : ((init -[on]-> state) ; (init -[off]-> init) ; (state -[on]->
    state) ; state -[off]-> init),flows : ((init [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; state [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : csys :(0,2),response : csys :(3,4),
    timer : 0,samplingIDs : csys,responseIDs : csys >),connections : (((csys ..
    on) --> (environment .. on)) ; ((csys .. off) --> (environment .. off)) ; (
    environment .. out) --> (csys .. temp)) >,400} | 500 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{initState,400} | 0 | infty} .
rewrites: 75344
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< csys : System | properties :
    none,features : (< on : DataOutPort | properties : none,content : * # [
    false] > < off : DataOutPort | properties : none,content : * # [false] > <
    temp : DataInPort | properties : none,content : [0.0] # [false],cache : [
    0.0] # [false] >),subcomponents : < cproc : Process | properties : none,
    features : (< on : DataOutPort | properties : none,content : * # [false] >
    < off : DataOutPort | properties : none,content : * # [false] > < temp :
    DataInPort | properties : none,content : [0.0] # [false],cache : [0.0] # [
    false] >),subcomponents : < cthread : Thread | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::SamplingTime => {{
    0.0 .. 2.0}}) ; HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features :
    (< on : DataOutPort | properties : none,content : * # [false] > < off :
    DataOutPort | properties : none,content : * # [false] > < temp : DataInPort
    | properties : none,content : [0.0] # [false],cache : [
    9.3880711162202189e+1] # [true] >),subcomponents : (< lower : Data |
    properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    9.3077183654263393e+1] # [true] >),connections : empty,varGen : <
    "top.cthread" >,currState : init,completeStates : init,variables : x :
    Real,transitions : init -[on dispatch]-> init {(v{x} := ((c[curr] + f[
    temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !)
    end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >,connections : ((
    temp --> (cthread .. temp)) ; ((cthread .. on) --> on) ; (cthread .. off)
    --> off) >,connections : ((temp --> (cproc .. temp)) ; ((cproc .. on) -->
    on) ; (cproc .. off) --> off) > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : csys > < off :
    EnvInPort | properties : none,content : * # [false],envCache : null(Unit),
    target : csys > < out : EnvOutPort | properties : none,content : [0.0] # [
    false],envCache : null(Real),target : csys >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [9.4412683881952546e+1] #
    [true] >,connections : x ==> out,varGen : < "top.environment" >,currMode :
    init,jumps : ((init -[on]-> state) ; (init -[off]-> init) ; (state -[on]->
    state) ; state -[off]-> init),flows : ((init [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; state [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : csys :(0,2),response : csys :(3,4),
    timer : 0,samplingIDs : csys,responseIDs : csys >),connections : (((csys ..
    on) --> (environment .. on)) ; ((csys .. off) --> (environment .. off)) ; (
    environment .. out) --> (csys .. temp)) >,800} | 500 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{initState,800} | 0 | infty} .
rewrites: 75396
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< csys : System | properties :
    none,features : (< on : DataOutPort | properties : none,content : * # [
    false] > < off : DataOutPort | properties : none,content : * # [false] > <
    temp : DataInPort | properties : none,content : [0.0] # [false],cache : [
    0.0] # [false] >),subcomponents : < cproc : Process | properties : none,
    features : (< on : DataOutPort | properties : none,content : * # [false] >
    < off : DataOutPort | properties : none,content : * # [false] > < temp :
    DataInPort | properties : none,content : [0.0] # [false],cache : [0.0] # [
    false] >),subcomponents : < cthread : Thread | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::SamplingTime => {{
    0.0 .. 2.0}}) ; HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features :
    (< on : DataOutPort | properties : none,content : * # [false] > < off :
    DataOutPort | properties : none,content : * # [false] > < temp : DataInPort
    | properties : none,content : [0.0] # [false],cache : [
    9.3043773667962213e+1] # [true] >),subcomponents : (< lower : Data |
    properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    9.2390812029053592e+1] # [true] >),connections : empty,varGen : <
    "top.cthread" >,currState : init,completeStates : init,variables : x :
    Real,transitions : init -[on dispatch]-> init {(v{x} := ((c[curr] + f[
    temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !)
    end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >,connections : ((
    temp --> (cthread .. temp)) ; ((cthread .. on) --> on) ; (cthread .. off)
    --> off) >,connections : ((temp --> (cproc .. temp)) ; ((cproc .. on) -->
    on) ; (cproc .. off) --> off) > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : csys > < off :
    EnvInPort | properties : none,content : * # [false],envCache : null(Unit),
    target : csys > < out : EnvOutPort | properties : none,content : [0.0] # [
    false],envCache : null(Real),target : csys >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [9.4039206802397771e+1] #
    [true] >,connections : x ==> out,varGen : < "top.environment" >,currMode :
    init,jumps : ((init -[on]-> state) ; (init -[off]-> init) ; (state -[on]->
    state) ; state -[off]-> init),flows : ((init [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; state [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : csys :(0,2),response : csys :(3,4),
    timer : 0,samplingIDs : csys,responseIDs : csys >),connections : (((csys ..
    on) --> (environment .. on)) ; ((csys .. off) --> (environment .. off)) ; (
    environment .. out) --> (csys .. temp)) >,1200} | 500 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{initState,1200} | 0 | infty} .
rewrites: 75337
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< csys : System | properties :
    none,features : (< on : DataOutPort | properties : none,content : * # [
    false] > < off : DataOutPort | properties : none,content : * # [false] > <
    temp : DataInPort | properties : none,content : [0.0] # [false],cache : [
    0.0] # [false] >),subcomponents : < cproc : Process | properties : none,
    features : (< on : DataOutPort | properties : none,content : * # [false] >
    < off : DataOutPort | properties : none,content : * # [false] > < temp :
    DataInPort | properties : none,content : [0.0] # [false],cache : [0.0] # [
    false] >),subcomponents : < cthread : Thread | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::SamplingTime => {{
    0.0 .. 2.0}}) ; HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features :
    (< on : DataOutPort | properties : none,content : * # [false] > < off :
    DataOutPort | properties : none,content : * # [false] > < temp : DataInPort
    | properties : none,content : [0.0] # [false],cache : [
    9.4196650230092246e+1] # [true] >),subcomponents : (< lower : Data |
    properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    9.3668497668111257e+1] # [true] >),connections : empty,varGen : <
    "top.cthread" >,currState : init,completeStates : init,variables : x :
    Real,transitions : init -[on dispatch]-> init {(v{x} := ((c[curr] + f[
    temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !)
    end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >,connections : ((
    temp --> (cthread .. temp)) ; ((cthread .. on) --> on) ; (cthread .. off)
    --> off) >,connections : ((temp --> (cproc .. temp)) ; ((cproc .. on) -->
    on) ; (cproc .. off) --> off) > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : csys > < off :
    EnvInPort | properties : none,content : * # [false],envCache : null(Unit),
    target : csys > < out : EnvOutPort | properties : none,content : [0.0] # [
    false],envCache : null(Real),target : csys >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [9.4999119325578107e+1] #
    [true] >,connections : x ==> out,varGen : < "top.environment" >,currMode :
    init,jumps : ((init -[on]-> state) ; (init -[off]-> init) ; (state -[on]->
    state) ; state -[off]-> init),flows : ((init [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; state [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : csys :(0,2),response : csys :(3,4),
    timer : 0,samplingIDs : csys,responseIDs : csys >),connections : (((csys ..
    on) --> (environment .. on)) ; ((csys .. off) --> (environment .. off)) ; (
    environment .. out) --> (csys .. temp)) >,1600} | 500 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{cinitState,0} | 0 | infty} .
rewrites: 63775
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< environment : EnvWithTimer |
    properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : csys > < off : EnvInPort | properties : none,content :
    * # [false],envCache : null(Unit),target : csys > < out : EnvOutPort |
    properties : none,content : [0.0] # [false],envCache : null(Real),target :
    csys >),subcomponents : < x : Data | properties : DataModel::InitialValue
    => {{16}},features : none,subcomponents : none,connections : empty,value :
    [9.5200193572053749e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : init,jumps : ((init -[on]-> state) ; (init
    -[off]-> init) ; (state -[on]-> state) ; state -[off]-> init),flows : ((
    init [x(t)= (c[x] + (v[t] * [[2.0000000000000001e-1]]))]) ; state [x(t)= (
    c[x] - (v[t] * [[1.0000000000000001e-1]]))]),sampling : csys :(0,2),
    response : csys :(3,4),timer : 0,samplingIDs : csys,responseIDs : csys > <
    csys . cproc . cthread : Thread | properties : ((TimingProperties::Period
    => {{5}}) ; (HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [9.4249841706512953e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{15}},features : none,subcomponents : none,
    connections : empty,value : [9.3596867381845982e+1] # [true] >),connections
    : empty,varGen : < "top.cthread" >,currState : init,completeStates : init,
    variables : x : Real,transitions : init -[on dispatch]-> init {(v{x} := ((
    c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >),
    connections : (((environment .. out) --> ((csys . cproc . cthread) ..
    temp)) ; (((csys . cproc . cthread) .. on) --> (environment .. on)) ; ((
    csys . cproc . cthread) .. off) --> (environment .. off)) >,400} | 500 |
    infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{cinitState,400} | 0 | infty} .
rewrites: 63774
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< environment : EnvWithTimer |
    properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : csys > < off : EnvInPort | properties : none,content :
    * # [false],envCache : null(Unit),target : csys > < out : EnvOutPort |
    properties : none,content : [0.0] # [false],envCache : null(Real),target :
    csys >),subcomponents : < x : Data | properties : DataModel::InitialValue
    => {{16}},features : none,subcomponents : none,connections : empty,value :
    [9.4412683881952546e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : init,jumps : ((init -[on]-> state) ; (init
    -[off]-> init) ; (state -[on]-> state) ; state -[off]-> init),flows : ((
    init [x(t)= (c[x] + (v[t] * [[2.0000000000000001e-1]]))]) ; state [x(t)= (
    c[x] - (v[t] * [[1.0000000000000001e-1]]))]),sampling : csys :(0,2),
    response : csys :(3,4),timer : 0,samplingIDs : csys,responseIDs : csys > <
    csys . cproc . cthread : Thread | properties : ((TimingProperties::Period
    => {{5}}) ; (HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [9.3880711162202189e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{15}},features : none,subcomponents : none,
    connections : empty,value : [9.3077183654263393e+1] # [true] >),connections
    : empty,varGen : < "top.cthread" >,currState : init,completeStates : init,
    variables : x : Real,transitions : init -[on dispatch]-> init {(v{x} := ((
    c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >),
    connections : (((environment .. out) --> ((csys . cproc . cthread) ..
    temp)) ; (((csys . cproc . cthread) .. on) --> (environment .. on)) ; ((
    csys . cproc . cthread) .. off) --> (environment .. off)) >,800} | 500 |
    infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{cinitState,800} | 0 | infty} .
rewrites: 63826
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< environment : EnvWithTimer |
    properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : csys > < off : EnvInPort | properties : none,content :
    * # [false],envCache : null(Unit),target : csys > < out : EnvOutPort |
    properties : none,content : [0.0] # [false],envCache : null(Real),target :
    csys >),subcomponents : < x : Data | properties : DataModel::InitialValue
    => {{16}},features : none,subcomponents : none,connections : empty,value :
    [9.4039206802397771e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : init,jumps : ((init -[on]-> state) ; (init
    -[off]-> init) ; (state -[on]-> state) ; state -[off]-> init),flows : ((
    init [x(t)= (c[x] + (v[t] * [[2.0000000000000001e-1]]))]) ; state [x(t)= (
    c[x] - (v[t] * [[1.0000000000000001e-1]]))]),sampling : csys :(0,2),
    response : csys :(3,4),timer : 0,samplingIDs : csys,responseIDs : csys > <
    csys . cproc . cthread : Thread | properties : ((TimingProperties::Period
    => {{5}}) ; (HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [9.3043773667962213e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{15}},features : none,subcomponents : none,
    connections : empty,value : [9.2390812029053592e+1] # [true] >),connections
    : empty,varGen : < "top.cthread" >,currState : init,completeStates : init,
    variables : x : Real,transitions : init -[on dispatch]-> init {(v{x} := ((
    c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >),
    connections : (((environment .. out) --> ((csys . cproc . cthread) ..
    temp)) ; (((csys . cproc . cthread) .. on) --> (environment .. on)) ; ((
    csys . cproc . cthread) .. off) --> (environment .. off)) >,1200} | 500 |
    infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM-HIE : {{cinitState,1200} | 0 | infty} .
rewrites: 63767
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< environment : EnvWithTimer |
    properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : csys > < off : EnvInPort | properties : none,content :
    * # [false],envCache : null(Unit),target : csys > < out : EnvOutPort |
    properties : none,content : [0.0] # [false],envCache : null(Real),target :
    csys >),subcomponents : < x : Data | properties : DataModel::InitialValue
    => {{16}},features : none,subcomponents : none,connections : empty,value :
    [9.4999119325578107e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : init,jumps : ((init -[on]-> state) ; (init
    -[off]-> init) ; (state -[on]-> state) ; state -[off]-> init),flows : ((
    init [x(t)= (c[x] + (v[t] * [[2.0000000000000001e-1]]))]) ; state [x(t)= (
    c[x] - (v[t] * [[1.0000000000000001e-1]]))]),sampling : csys :(0,2),
    response : csys :(3,4),timer : 0,samplingIDs : csys,responseIDs : csys > <
    csys . cproc . cthread : Thread | properties : ((TimingProperties::Period
    => {{5}}) ; (HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [9.4196650230092246e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{15}},features : none,subcomponents : none,
    connections : empty,value : [9.3668497668111257e+1] # [true] >),connections
    : empty,varGen : < "top.cthread" >,currState : init,completeStates : init,
    variables : x : Real,transitions : init -[on dispatch]-> init {(v{x} := ((
    c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]},loopBound : 0,transBound : 2 >),
    connections : (((environment .. out) --> ((csys . cproc . cthread) ..
    temp)) ; (((csys . cproc . cthread) .. on) --> (environment .. on)) ; ((
    csys . cproc . cthread) .. off) --> (environment .. off)) >,1600} | 500 |
    infty}
Bye.
